src/components/application_manager/include/application_manager/core_service.h

```diff
@@ -0,0 +1,257 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_CORE_SERVICE_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_CORE_SERVICE_H_
+
+#include <string>
+#include <vector>
+#include "application_manager/service.h"
+#include "application_manager/application.h"
+#include "policy/policy_types.h"
+
+namespace Json {
+class Value;
+}
+
+namespace application_manager {
+
+struct CommandParametersPermissions;
+
+/**
+ * @brief Class through which the plug-in can interact with the core
+ */
+class CoreService : public Service {
+ public:
+  /**
+   * @brief CoreService class destructor
+   */
+  explicit CoreService(ApplicationManager& application_manager);
+
+  /**
+   * @brief CoreService class destructor
+   */
+  virtual ~CoreService();
+
+  /**
+   * @brief Checks message permissions and cuts parameters according
+   * to policy table permissions
+   * @param msg message to cut disallowed parameters
+   * @return result according by mobile API
+   */
+  mobile_apis::Result::eType CheckPolicyPermissions(MessagePtr msg) FINAL;
+
+  /**
+   * Checks access to requested equipment of vehicle
+   * @param app_id id of application
+   * @param module type
+   * @param rpc name of rpc
+   * @param params parameters list
+   * @return return allowed if access exist,
+   * manual if need to send question to driver otherwise disallowed
+   */
+  TypeAccess CheckAccess(const ApplicationId& app_id,
+                         const std::string& module,
+                         const std::string& rpc,
+                         const std::vector<std::string>& params) FINAL;
+
+  /**
+   * Checks access to module for application
+   * @param app_id id of application
+   * @param module type
+   * @return true if module is allowed for application
+   */
+  bool CheckModule(const ApplicationId& app_id,
+                   const std::string& module) FINAL;
+
+  /**
+   * Sets access to functional group which contains given RPC for application
+   * @param app_id id of application
+   * @param module type
+   * @param allowed true if driver has given access
+   */
+  void SetAccess(const ApplicationId& app_id,
+                 const std::string& module,
+                 bool allowed) FINAL;
+
+  /**
+   * Resets access by group name for all applications
+   * @param group_name group name
```

Comment:
I see parameter "const ApplicationId& app_id" not "group name", please update

Reply:
Confirmed resolved 👍